Content.Shared/Body/Systems/SharedBodySystem.Body.cs

```diff
@@ -9,120 +9,95 @@
 using Robust.Shared.Containers;
 using Robust.Shared.GameStates;
 using Robust.Shared.Network;
+using MapInitEvent = Robust.Shared.GameObjects.MapInitEvent;
 
 namespace Content.Shared.Body.Systems;
 
 public partial class SharedBodySystem
 {
     [Dependency] private readonly INetManager _netManager = default!;
-
     public void InitializeBody()
     {
+        SubscribeLocalEvent<BodyComponent, MapInitEvent>(OnBodyMapInit);
+        SubscribeLocalEvent<BodyComponent, CanDragEvent>(OnBodyCanDrag);
         SubscribeLocalEvent<BodyComponent, ComponentInit>(OnBodyInit);
-
         SubscribeLocalEvent<BodyComponent, ComponentGetState>(OnBodyGetState);
         SubscribeLocalEvent<BodyComponent, ComponentHandleState>(OnBodyHandleState);
-        SubscribeLocalEvent<BodyComponent, CanDragEvent>(OnBodyCanDrag);
     }
 
-    private void OnBodyCanDrag(EntityUid uid, BodyComponent component, ref CanDragEvent args)
+    private void OnBodyHandleState(EntityUid uid, BodyComponent component, ref ComponentHandleState args)
     {
-        args.Handled = true;
+        if (args.Current is not BodyComponentState state)
+            return;
+        component.Prototype = state.Prototype;
+        component.RootContainer = Containers.EnsureContainer<ContainerSlot>(uid, state.ContainerId);
+        component.RootPartSlot = state.RootPartSlot;
+        component.GibSound = state.GibSound;
+        component.RequiredLegs = state.RequiredLegs;
+        component.LegEntities = EntityManager.GetEntitySet(state.LegNetEntities);
     }
```

Comment:
any reason this cant be auto generated? legentities conversion should get handled fine

Reply:
It doesn't get handled automatically (NetEntities is a fuck), that's why I need to handle this manually